x/auth/legacy/legacytx/stdtx.go

```diff
@@ -16,10 +16,9 @@ import (
 
 // Interface implementation checks
 var (
-	_ sdk.Tx             = (*StdTx)(nil)
-	_ codectypes.IntoAny = (*StdTx)(nil)
-	_ sdk.TxWithMemo     = (*StdTx)(nil)
-	_ sdk.FeeTx          = (*StdTx)(nil)
+	_ sdk.Tx         = (*StdTx)(nil)
+	_ sdk.TxWithMemo = (*StdTx)(nil)
+	_ sdk.FeeTx      = (*StdTx)(nil)
```

Comment:
shouldn't StdTx also implement ProtoTxProvider?

TxResponse has a tx field of type Any to handle StdTx too, afaiu. It would be good to add a test (or maybe it already exists) to:
- send a StdTx via the REST endpoint
- query the tx by hash
- decode the Any field on TxResponse

Reply:
`ProtoTxProvider` should return a proto transaction - it's used to make sure that we can provide a proto transaction. Initially I was thinking to add that to the `TxBuilder` interface, but I realized that `StdTx` needs to be different. Am I missing something?